climakitae/data_loaders.py

```diff
@@ -148,42 +134,158 @@ def set_subarea(boundary_dataset):
+            if selections.append_historical and "ssp" in scenario:
+
+                # Reset name
+                da_name = "Historical + " + _convert_scenario(scenario, reverse = True)
+
+                # Get filenames
+                try:
+                    historical_filename = [name for name in dsets.keys() if simulation + "." + "historical" in name][0]
+                    ssp_filename = [name for name in dsets.keys() if simulation + "." + scenario in name][0]
+                except: # Some simulation + ssp options are not available. Just continue with the loop if no filename is found
+                    continue
+                # Grab data
+                historical_data = dsets[historical_filename][selections.variable_id]
+                ssp_data = dsets[ssp_filename][selections.variable_id]
+
+                # Concatenate data. Rename scenario attribute
+                historical_appended = xr.concat(
+                    [historical_data, ssp_data],
+                    dim = "time",
+                    coords = 'minimal',
+                    compat = 'override',
+                    join = 'inner'
+                )
+                sim_list.append(historical_appended)
+
+            else:
+                try:
+                    filename = [name for name in dsets.keys() if simulation + "." + scenario in name][0]
+                except:
+                    continue
+                sim_list.append(dsets[filename][selections.variable_id])
+
+        # Concatenate along simulation dimension
+        da = xr.concat(
+            sim_list,
+            dim = "simulation",
+            coords = 'minimal',
+            compat = 'override'
+        )
+        da_list.append(da.assign_coords({"scenario": da_name}))
+
+    # Concatenate along scenario dimension
+    da_final = xr.concat(
+        da_list,
+        dim = "scenario",
+        coords = 'minimal',
+        compat = 'override'
+    )
+
+    # Rename
+    da_final.name = selections.variable
+
+    # Add attributes
+    orig_attrs = dsets[list(dsets.keys())[0]].attrs
+    da_final.attrs = { # Add descriptive attributes to DataArray
+        #"conventions": orig_attrs["conventions"],
+        "institution": orig_attrs["institution"],
+        "source": orig_attrs["source"],
+        "resolution": selections.resolution,
+        "frequency": selections.timescale,
+        "grid_mapping": da_final.attrs["grid_mapping"],
+        "variable_id": selections.variable_id,
+        #"long_name": da_final.attrs["long_name"],
+        "extended_description": selections.extended_description,
+        "units": da_final.attrs["units"]
+    }
+    return da_final
+
+# ============ Read from catalog function used by ck.Application ===============
+
+def _read_from_catalog(selections, location, cat):
+    """
+    The primary and first data loading method, called by
+    core.Application.generate, it returns a DataArray (which can be quite large)
```

Comment:
`generate` is now `retrieve` right?

Reply:
yes, i will update that comment. i had just copy/pasted